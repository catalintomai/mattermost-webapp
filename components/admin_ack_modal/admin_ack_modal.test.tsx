// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {Modal} from 'react-bootstrap';
import {UserProfile} from 'mattermost-redux/src/types/users';

import AdminAckModal from 'components/admin_ack_modal/admin_ack_modal';

jest.mock('react-dom', () => ({
    findDOMNode: () => ({
        blur: jest.fn(),
    }),
}));

describe('components/AdminAckModal', () => {
    const serverError = 'some error';

    const baseProps = {
        user: {
            id: 'someUserId',
            first_name: 'Fake',
            last_name: 'Person',
            email: 'a@test.com'
        } as UserProfile,
        show: false,
        closeParentComponent: jest.fn(),
        actions: {
            closeModal: jest.fn(),
            sendAdminAck: jest.fn(),
        },
    };

    test('should match snapshot, init', () => {
        const wrapper = shallow<AdminAckModal>(
            <AdminAckModal {...baseProps}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('error display', () => {
        const wrapper = shallow<AdminAckModal>(
            <AdminAckModal {...baseProps}/>
        );

        wrapper.setState({serverError});
        expect(wrapper).toMatchSnapshot();
    });

    test('should match state when onHide is called', () => {
        const wrapper = shallow<AdminAckModal>(
            <AdminAckModal {...baseProps}/>
        );

        wrapper.setState({saving: true});
        wrapper.instance().onHide();
        expect(wrapper.state('saving')).toEqual(false);
    });

    test('should match state when onHideWithParent is called', () => {
        const wrapper = shallow<AdminAckModal>(
            <AdminAckModal {...baseProps}/>
        );

        wrapper.setState({saving: true});
        wrapper.instance().onHideWithParent();

        expect(baseProps.closeParentComponent).toHaveBeenCalledTimes(1);
        expect(wrapper.state('saving')).toEqual(false);
    });

    test('send ack on submit button click', () => {
        const wrapper = shallow<AdminAckModal>(
            <AdminAckModal {...baseProps}/>
        );

        wrapper.setState({saving: true});
        wrapper.find('.save-button').simulate('click');
        expect(baseProps.actions.sendAdminAck).toHaveBeenCalledTimes(1);

        expect(baseProps.closeParentComponent).toHaveBeenCalledTimes(1);
        expect(wrapper.state('saving')).toEqual(false);
    });

    test('should have called props.onHide when Modal.onExited is called', () => {
        const props = {...baseProps};
        const wrapper = shallow(
            <AdminAckModal {...props}/>
        );

        wrapper.find(Modal).props().onExited!(document.createElement('div'));
        expect(baseProps.actions.closeModal).toHaveBeenCalledTimes(1);
    });
});