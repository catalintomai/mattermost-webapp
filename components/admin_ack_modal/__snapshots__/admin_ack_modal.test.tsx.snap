// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AdminAckModal error display 1`] = `
<Modal
  animation={true}
  aria-labelledby="adminAckHeaderModalLabel"
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="a11y__modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={false}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onExited={[Function]}
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  role="dialog"
  show={false}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h1"
      id="adminAckHeaderModalLabel"
    >
      <FormattedMessage
        defaultMessage="Warning!"
        id="admin_ack_modal.header.title"
        values={Object {}}
      />
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div
      className="alert alert-danger"
    >
      <FormattedMessage
        defaultMessage="The number of active users is greater than the supported limit. Let us know that you acknowledge the issue by clicking the Submit button!"
        id="admin_ack_modal.description"
        values={Object {}}
      />
      <br />
      <div
        className="form-group has-error"
      >
        <br />
        <label
          className="control-label"
        >
          <FormattedMessage
            defaultMessage="Click {link} to acknowledge the warning!"
            id="admin_ack_modal.mailto.message"
            values={
              Object {
                "link": <AdminAckErrorLink
                  defaultMessage="MailTo"
                  messageId="admin_ack_modal.mailto.message.link"
                  onClickHandler={[Function]}
                  url="mailto:support@mattermost.com?cc=a@test.com&subject=Warning: Number of active users exceeded the limit&body=Number of active users exceeded the limit SiteURL: http://localhost:8065 Contact Email: a@test.com"
                />,
              }
            }
          />
        </label>
      </div>
    </div>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <button
      className="btn btn-link cancel-button"
      onClick={[Function]}
      type="button"
    >
      <FormattedMessage
        defaultMessage="Cancel"
        id="admin_ack_modal.cancel"
        values={Object {}}
      />
    </button>
    <button
      autoFocus={true}
      className="btn btn-danger save-button"
      data-dismiss="modal"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <FormattedMessage
        defaultMessage="Acknowledge"
        id="admin_ack_modal.submit"
        values={Object {}}
      />
    </button>
  </ModalFooter>
</Modal>
`;

exports[`components/AdminAckModal should match snapshot, init 1`] = `
<Modal
  animation={true}
  aria-labelledby="adminAckHeaderModalLabel"
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="a11y__modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={false}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onExited={[Function]}
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  role="dialog"
  show={false}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h1"
      id="adminAckHeaderModalLabel"
    >
      <FormattedMessage
        defaultMessage="Warning!"
        id="admin_ack_modal.header.title"
        values={Object {}}
      />
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div
      className="alert alert-danger"
    >
      <FormattedMessage
        defaultMessage="The number of active users is greater than the supported limit. Let us know that you acknowledge the issue by clicking the Submit button!"
        id="admin_ack_modal.description"
        values={Object {}}
      />
      <br />
    </div>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <button
      className="btn btn-link cancel-button"
      onClick={[Function]}
      type="button"
    >
      <FormattedMessage
        defaultMessage="Cancel"
        id="admin_ack_modal.cancel"
        values={Object {}}
      />
    </button>
    <button
      autoFocus={true}
      className="btn btn-danger save-button"
      data-dismiss="modal"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <FormattedMessage
        defaultMessage="Acknowledge"
        id="admin_ack_modal.submit"
        values={Object {}}
      />
    </button>
  </ModalFooter>
</Modal>
`;
